src/SDKs/Batch/DataPlane/Tools/ObjectModelCodeGeneration/ObjectModelCodeGenerator/ObjectModelCodeGenerator.csproj

```diff
@@ -0,0 +1,88 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <PackageId>ObjectModelCodeGenerator</PackageId>
+    <Description>Azure Batch object model code generator</Description>
+    <!--After fixing all warnings, enable this option-->
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
+    <AssemblyName>ObjectModelCodeGenerator</AssemblyName>
+    <OutputType>Exe</OutputType>
+    <RestorePackagesPath>$(LibraryNugetPackageFolder)</RestorePackagesPath>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>net46</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.VisualStudio.TextTemplating.14.0" Version="14.3.25407" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\CodeGenerationLibrary\CodeGenerationLibrary.csproj" />
+    <ProjectReference Include="..\RoslynParser\ProxyLayerParser.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="CollectionConversionSnippet.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>CollectionConversionSnippet.cs</LastGenOutput>
+    </None>
+    <None Update="GetTransportObjectSnippet.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>GetTransportObjectSnippet.cs</LastGenOutput>
+    </None>
+    <None Update="ModelClassTemplate.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModelClassTemplate.cs</LastGenOutput>
+    </None>
+    <None Update="ModifiableClassTemplate.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>ModifiableClassTemplate.cs</LastGenOutput>
+    </None>
+    <None Update="NamedBatchRequests.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>NamedBatchRequests.cs</LastGenOutput>
+    </None>
+    <None Update="StaticReadOnlyClassTemplate.tt">
+      <Generator>TextTemplatingFilePreprocessor</Generator>
+      <LastGenOutput>StaticReadOnlyClassTemplate.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Compile Update="CollectionConversionSnippet.cs">
```

Comment:
No, we need these entries because these are cs files genereated from T4 text templates. I didn't add this to this file manually, VS did when I added the T4 text template files (.tt), so i assume that VS needs it (probably for the `DesignTime` and `AutoGen` annotations